application/src/main/java/org/opentripplanner/model/plan/walkstep/WalkStep.java

```diff
@@ -46,7 +48,13 @@ public final class WalkStep {
   private final boolean walkingBike;
 
   private final String highwayExit;
+
+  @Nullable
   private final Entrance entrance;
+
+  @Nullable
+  private final VerticalTransportationUse verticalTransportationUse;
```

Comment:
In the API it is only allowed to have either an entrence or a verticalTransportationUse, this bussiness rule is not enforced here.

The other problem here is that the Itinerary/Leg/WaleStep is decorated with information with in many cases is not used. If the information is pure decoration, then this should happen in the API using shared builder/fetching logic (service for data-fetchers).